x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/operator/DefaultOperatorOnlyRegistry.java

```diff
@@ -0,0 +1,113 @@
+import org.elasticsearch.action.admin.cluster.settings.ClusterUpdateSettingsRequest;
+import org.elasticsearch.common.Strings;
+import org.elasticsearch.common.settings.ClusterSettings;
+import org.elasticsearch.common.settings.Setting;
+import org.elasticsearch.license.DeleteLicenseAction;
+import org.elasticsearch.license.PutLicenseAction;
+import org.elasticsearch.rest.RestHandler;
+import org.elasticsearch.rest.RestRequest;
+import org.elasticsearch.rest.RestResponse;
+import org.elasticsearch.transport.TransportRequest;
+
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Stream;
+
+public class DefaultOperatorOnlyRegistry implements OperatorOnlyRegistry {
+
+    public static final Set<String> SIMPLE_ACTIONS = Set.of(
+        AddVotingConfigExclusionsAction.NAME,
+        ClearVotingConfigExclusionsAction.NAME,
+        PutLicenseAction.NAME,
+        DeleteLicenseAction.NAME,
+        // Autoscaling does not publish its actions to core, literal strings are needed.
+        "cluster:admin/autoscaling/put_autoscaling_policy",
+        "cluster:admin/autoscaling/delete_autoscaling_policy",
+        // Repository analysis actions are not mentioned in core, literal strings are needed.
+        "cluster:admin/repository/analyze",
+        "cluster:admin/repository/analyze/blob",
+        "cluster:admin/repository/analyze/blob/read",
+        "cluster:admin/repository/analyze/register",
+        // Node shutdown APIs are operator only
+        "cluster:admin/shutdown/create",
+        "cluster:admin/shutdown/get",
+        "cluster:admin/shutdown/delete",
+        // Node removal prevalidation API
+        PrevalidateNodeRemovalAction.NAME,
+        // Desired Nodes API
+        DeleteDesiredNodesAction.NAME,
+        GetDesiredNodesAction.NAME,
+        UpdateDesiredNodesAction.NAME,
+        GetDesiredBalanceAction.NAME,
+        DeleteDesiredBalanceAction.NAME
+    );
+
+    private final ClusterSettings clusterSettings;
+
+    public DefaultOperatorOnlyRegistry(ClusterSettings clusterSettings) {
+        this.clusterSettings = clusterSettings;
+    }
+
+    /**
+     * Check whether the given action and request qualify as operator-only. The method returns
+     * null if the action+request is NOT operator-only. Other it returns a violation object
+     * that contains the message for details.
+     */
+    public OperatorPrivilegesViolation check(String action, TransportRequest request) {
+        if (SIMPLE_ACTIONS.contains(action)) {
+            return () -> "action [" + action + "]";
+        } else if (ClusterUpdateSettingsAction.NAME.equals(action)) {
+            assert request instanceof ClusterUpdateSettingsRequest;
+            return checkClusterUpdateSettings((ClusterUpdateSettingsRequest) request);
+        } else {
+            return null;
+        }
+    }
+
+    @Override
+    public RestResponse checkRestFull(RestHandler restHandler) {
```

Comment:
the filename, this method (checkRestFull) and checkRestPartial are the only changes to this class. The original name (without Default) is now an interface.